src/hooks/useAuthorization.ts

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { cancellableFetchPOST } from '../cancellable-fetch';
+
+export enum AuthorizationStatus {
+  Authorized = 'authorized',
+  AuthorizedError = 'authorizedError',
+  AuthorizedLoading = 'authorizedLoading',
+}
+
+const AUTHORIZATION_ENDPOINT = '/api/proxy/plugin/lightspeed-console-plugin/ols/authorized';
+
+type AuthorizationResponse = {
+  user_id: string;
+  username: string;
+};
+
+export const useAuthorization = (): [AuthorizationStatus] => {
+  const [authorizationStatus, setAuthorizationStatus] = React.useState<AuthorizationStatus>(
+    AuthorizationStatus.AuthorizedLoading,
+  );
+
+  React.useEffect(() => {
+    const checkAuthorization = async () => {
+      console.log('checking authorization');
+      const headers = {
+        'Content-Type': 'application/json',
```

Comment:
We shouldn't need to pass this header, since it's already added in `cancellableFetchPOST`

Reply:
fixed